src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: () => void,
+  toggleStatusTodo: () => void,
+  renameTodo: (todo: Todo, newTitle: string) => void,
+  loading: { todoId: number, isLoading: boolean },
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  toggleStatusTodo,
+  renameTodo,
+  loading,
+}) => {
+  const [isRenamed, setIsRenamed] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!newTitle) {
+      deleteTodo();
+
+      return;
+    }
+
+    if (newTitle !== todo.title) {
+      renameTodo(todo, newTitle);
+    }
+
+    setIsRenamed(false);
+  };
+
+  const reset = (event: React.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      handleSubmit(event);
+    }
+  };
+
+  return (
+    <div
+      className={classNames(
+        'todo',
+        { completed: todo.completed },
+      )}
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          onClick={toggleStatusTodo}
+        />
+      </label>
+
+      {isRenamed ? (
+        <form
+          onSubmit={handleSubmit}
+        >
+          <input
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value={newTitle}
+            onChange={handleChange}
+            // eslint-disable-next-line jsx-a11y/no-autofocus
```

Comment:
Remake autofocus by using `ref`.

[More info](https://uk.reactjs.org/docs/hooks-reference.html#useref)

Reply:
Try to avoid linter comments.